test/qa/test_combinations.py

```diff
@@ -156,3 +156,70 @@ def test_connect_country_and_city(country, city):
     print(output)
     assert lib.is_disconnect_successful(output)
     assert network.is_disconnected()
+
+
+@pytest.mark.parametrize("target_tech,target_proto,target_obfuscated", lib.STANDARD_TECHNOLOGIES)
+@pytest.mark.parametrize("source_tech,source_proto,source_obfuscated", lib.STANDARD_TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(40)
+def test_status_change_technology_and_protocol(
+    source_tech,
+    target_tech,
+    source_proto,
+    target_proto,
+    source_obfuscated,
+    target_obfuscated,
+):
+    lib.set_technology_and_protocol(source_tech, source_proto, source_obfuscated)
+
+    sh.nordvpn.connect()
+    assert source_tech.upper() in sh.nordvpn.status()
+
+    if source_tech == "openvpn":
+        assert source_proto.upper() in sh.nordvpn.status()
+    else:
+        assert "UDP" in sh.nordvpn.status()
+
+    lib.set_technology_and_protocol(target_tech, target_proto, target_obfuscated)
+    assert source_tech.upper() in sh.nordvpn.status()
+
+    if source_tech == "openvpn":
+        assert source_tech.upper() in sh.nordvpn.status()
+    else:
+        assert "UDP" in sh.nordvpn.status()
+
+    output = sh.nordvpn.disconnect()
+    print(output)
+    assert lib.is_disconnect_successful(output)
+    assert network.is_disconnected()
+
+
+@pytest.mark.parametrize("target_tech,target_proto,target_obfuscated", lib.STANDARD_TECHNOLOGIES)
+@pytest.mark.parametrize("source_tech,source_proto,source_obfuscated", lib.STANDARD_TECHNOLOGIES)
+@pytest.mark.flaky(reruns=2, reruns_delay=90)
+@timeout_decorator.timeout(40)
+def test_status_change_technology_and_protocol_reconnect(
+    source_tech,
+    target_tech,
+    source_proto,
+    target_proto,
+    source_obfuscated,
+    target_obfuscated,
+):
+    lib.set_technology_and_protocol(source_tech, source_proto, source_obfuscated)
+
+    sh.nordvpn.connect()
+
+    lib.set_technology_and_protocol(target_tech, target_proto, target_obfuscated)
+    sh.nordvpn.connect()
```

Comment:
I'm not 100% sure, but if test fails on this connect, then on a rerun it might fail again on the first connect because we are already connected (I'm not sure whether after second failed connect we would be left disconnected or still connected to the first server). See this test for example how to avoid this using `lib.defer` https://github.com/NordSecurity/nordvpn-linux/blob/main/test/qa/test_dns.py#L43
In your test above it's not strictly necessary since there's only one connect and it's at the start of the test. On the other hand I would still personally recommend to do this `lib.defer(disconnect)` before every connect in every test just in case, but it's your choice.